backend/api/office_hours/oh_section.py

```diff
@@ -0,0 +1,92 @@
+"""OH Section API
+
+This API is used to access OH section data."""
+
+from fastapi import APIRouter, Depends
+
+from backend.models.office_hours.oh_section import OfficeHoursSectionDraft
+from backend.models.office_hours.oh_section_details import OfficeHoursSectionDetails
+from ..authentication import registered_user
+from ...models import User
+
+
+__authors__ = ["Sadie Amato", "Bailey DeSouza"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+api = APIRouter(prefix="/api/office-hours/section")
+openapi_tags = {
+    "name": "Office Hours",
+    "description": "Office hours section functionality",
+}
+
+
+@api.post("", response_model=OfficeHoursSectionDetails, tags=["Office Hours"])
+def new_oh_section(
+    oh_section: OfficeHoursSectionDraft,
+    subject: User = Depends(registered_user),
+    oh_section_service: OfficeHoursSectionService = Depends(),
+) -> OfficeHoursSectionDetails:
+    """
+    Adds a new OH section to the database
+
+    Returns:
+        OfficeHoursSectionDetails: OH Section created
+    """
+    return oh_section_service.create(subject, oh_section)
+
+
+@api.get("/{term_id}", response_model=list[OfficeHoursSectionDetails], tags=["Office Hours"])
```

Comment:
If we call this endpoint, then our api would be something like `"/office-hours/section/{term_id}"`. I believe this is typically the convention if we wanted to get an office hours section by an id -- this path could be misleading as I would think the id in the last part of the url would represent a section id rather than term id.

Since we are trying to query from a category such as term, maybe we could add a term before the term id so it's more apparent that the end point if querying by term id and not a section id!

Maybe something like this, but we could discuss more about how it should look!
`"/office-hours/section/term/{term_id}"`

Reply:
Agree with this concern. Could also make the route feel like a child of a term route if willing to modify academics

Reply:
This is a good point, thank you! I've changed the endpoint to be "/office-hours/section/{term_id}" and chose to not modify academics (though we can in the future if we want to change that direction).